File: app/src/main/java/edu/uark/cartapp/models/api/fields/ProductListingFieldName.java
package edu.uark.cartapp.models.api.fields;

import edu.uark.cartapp.models.api.interfaces.FieldNameInterface;

/* ==== APP ProductListingFieldName.java ====*/
public enum ProductListingFieldName implements FieldNameInterface {
	PRODUCTS();
	private String fieldName;

	ProductListingFieldName() {
		this.fieldName = "products";
	}

	public String getFieldName() { return this.fieldName; }
}

File: app/src/main/java/edu/uark/cartapp/models/api/fields/ProductListingFieldName.kt
package edu.uark.cartapp.models.api.fields

import edu.uark.cartapp.models.api.interfaces.FieldNameInterface

/* ==== APP ProductListingFieldName.java ====*/
enum class ProductListingFieldName private constructor() : FieldNameInterface {
	PRODUCTS;

	private val fieldName: String

	init {
		this.fieldName = "products"
	}

	override fun getFieldName(): String = this.fieldName

}